### PR TITLE
Added RN docs for DATADOG_RELEASE_VERSION env var

### PR DESCRIPTION
**Pull request by marco-saia-datadog**
### What does this PR do? What is the motivation?
Adds documentation for new environment variable that can be used to override the release version of the uploaded sourcemaps on Android and iOS.

### Merge instructions

- [x] Waiting for Datadog React Native v2.3.5 to be released (https://github.com/DataDog/dd-sdk-reactnative/pull/668)

### CODE REVIEW

File: content/en/real_user_monitoring/error_tracking/mobile/reactnative.md [hunk 1]
```diff
@@ -85,6 +85,12 @@ project.ext.datadog = [
 
 Options for the `datadog-ci react-native xcode` command are available on the [command documentation page][12].
 
+#### Custom Release Version
```
Comment (drichards-87) [inline content/en/real_user_monitoring/error_tracking/mobile/reactnative.md:88]:
```suggestion
#### Specifying a custom release version
```

**Review by drichards-87 — CHANGES_REQUESTED** [review]
Left a small suggestion.

**Comment by drichards-87** [comment]
@marco-saia-datadog Feel free to message me or post in the documentation Slack channel when this is ready to be merged. Thanks!

**Comment by marco-saia-datadog** [comment]
@drichards-87 It is ready to be merged now, thank you!

**Review by drichards-87 — APPROVED** [review]

**Pull request merged**